### PR TITLE
svdesu/ch145/edittext-underline

### PR DESCRIPTION
**Pull request by SVD6**
Story: https://app.clubhouse.io/chrono/story/145/edittext-underline

What it looks like now:
image

### CODE REVIEW

File: app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreate.kt [hunk 1]
```diff
@@ -29,90 +27,73 @@ private const val TIME_CHANGE_VALUE: Int = 5
...
                 saveCircuit()
             }
         }
 
-        bind!!.circuitName.addTextChangedListener {
-            if (bind!!.circuitName.length() <= 2) {
-                bind!!.warning.text = getString(R.string.min_char_warning)
-                bind!!.warning.visibility = View.VISIBLE
+        bind.circuitName.addTextChangedListener {
+            if (bind.circuitName.length() <= 2 || bind.circuitName.length() >= 50) {
```
Comment (harigovind1998) [inline app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreate.kt]: Dont use magic numbers. Define 2 and 50 as a const.

File: app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreate.kt [hunk 2]
```diff
@@ -125,29 +106,29 @@ class CircuitCreate : BaseActivity() {
     }
 
     private fun validateInputs(): Boolean {
-        if (bind!!.circuitName.text.toString() == "") {
+        if (bind.circuitName.text.toString() == "") {
```
Comment (harigovind1998) [inline app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreate.kt]: we should be using isEmpty(). Sorry i think i let this slip when i was in here last.

File: app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreate.kt [hunk 3]
```diff
@@ -125,29 +106,29 @@ class CircuitCreate : BaseActivity() {
...
             Toast.makeText(this, "Please enter a circuit name", Toast.LENGTH_SHORT).show()
             return false
         }
-        if (!(bind!!.setNum.text.toString().matches(("^[1-9]\\d*\$").toRegex()))) {
+        if (!(bind.setNum.text.toString().matches(("^[1-9]\\d*\$").toRegex()))) {
```
Comment (harigovind1998) [inline app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreate.kt:114]: We can probably create a const for the regex string.

File: app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreate.kt (hunk 2 above)
Comment (SVD6) [reply app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreate.kt]: Done

File: app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreate.kt (hunk 1 above)
Comment (SVD6) [reply app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreate.kt]: Done, except 2 isn't there anymore

File: app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreate.kt (hunk 3 above)
Comment (SVD6) [reply app/src/main/java/ca/chronofit/chrono/circuit/CircuitCreate.kt:114]: I'm gonna leave it for now, I'm gonna add constants in another story and we can update everything later on

**Review by harigovind1998 — APPROVED** [review]

**Pull request merged**